source/brailleDisplayDrivers/seikantk.py

```diff
@@ -286,16 +297,15 @@ def __init__(self, index):
 		self.routingIndex = index
 
 
-def _getKeyNames(keys: int) -> Set[int]:
-	return {_keyNames[1 << i] for i in range(16) if (1 << i) & keys}
-
-
-def _getDotNames(dots: int) -> Set[int]:
-	return {_dotNames[1 << i] for i in range(8) if (1 << i) & dots}
+def _getKeyNames(keys: int, names: Dict[int, str]) -> Set[str]:
+	"""Converts a bitset of hardware buttons and keys to their equivalent names"""
+	return {keyName for bitFlag, keyName in names.items() if bitFlag & keys}
 
 
-def _getRoutingIndexes(routingKeys: bytes) -> Set[int]:
-	return {i * 8 + j for i in range(len(routingKeys)) for j in range(8) if routingKeys[i] & (1 << j)}
+def _getRoutingIndexes(routingKeys: bytes, numKeys: int) -> Set[int]:
+	"""Converts a bitset of routing keys to their 0-index, up to 15 or 39 depending on the device"""
+	routingKeyBitSet = sum([routingKeys[i] << (8 * i) for i in range(len(routingKeys))])
```

Comment:
Why use `range(len(` over `enumerate`, in both cases this needs the index and value and visits every item?